### PR TITLE
Adding couple of unit tests for creating operators.

### PR DESCRIPTION
**Pull request by Jiri-Kremser**
Hey, I am playing with Rx in Ruby land, so I've cloned the repo and add couple of unit tests.

The 2 tests for repeat are commented out, because the internal clock of the very last event is off (`on_completed` has the same time as the last `on_next`). I think, it's because of not using the `scheduler.schedule_recursive{_with_state}`, perhaps they should be rewritten to use it. Not sure about the performance impact, though.

### CODE REVIEW

File: test/rx_ruby/core/test_observable_creation.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Copyright (c) Microsoft Open Technologies, Inc. All rights reserved. See License.txt in the project root for license information.
 
-require 'test_helper'
+require "#{File.dirname(__FILE__)}/../../test_helper"
```
Comment (Jiri-Kremser) [inline test/rx_ruby/core/test_observable_creation.rb]: This is not necessary, but it allows for debugging in RubyMine and running the tests with gem called `m`.

File: test/rx_ruby/core/test_observable_creation.rb [hunk 2]
```diff
@@ -463,21 +462,179 @@ def range_dispose
...
+
+    msgs = [
+        on_next(201, 42),
+        on_completed(202)
+    ]
+    assert_messages msgs, res.messages
   end
 =end
 
+  def test_repeat_infinitely_dispose
```
Comment (minamimonji) [inline test/rx_ruby/core/test_observable_creation.rb]: This test fails on Travis CI. Can you take a look at this?
https://travis-ci.org/ReactiveX/RxRuby/jobs/118394732

Comment (Jiri-Kremser) [inline test/rx_ruby/core/test_observable_creation.rb]: @minamimonji The previous code was good according to the Ruby [spec](https://github.com/ruby/spec/blob/master/language/return_spec.rb). It looks like, the Rubinius is a little bit special :) This change should not change the semantics of the code, though.

Comment (minamimonji) [inline test/rx_ruby/core/test_observable_creation.rb]: @Jiri-Kremser Thank you. LGTM :+1:

**Pull request merged**